src/main/java/com/burgstaller/okhttp/AuthenticationCacheInterceptor.java

```diff
@@ -50,15 +46,28 @@ public Response intercept(Chain chain) throws IOException {
 
         // Cached response was used, but it produced unauthorized response (cache expired).
         int responseCode = response != null ? response.code() : 0;
-        if (authenticator != null && (responseCode == HTTP_UNAUTHORIZED || responseCode == HTTP_PROXY_AUTH)) {
-            // Remove cached authenticator and resend request
-            if (authCache.remove(key) != null) {
-                response.body().close();
-                Platform.get().log("Cached authentication expired. Sending a new request.", Platform.INFO, null);
-                // Force sending a new request without "Authorization" header
+
+        if (authenticator != null && (
+                (cacheKeyProvider.applyToProxy() && responseCode == HTTP_PROXY_AUTH) ||
+                        !cacheKeyProvider.applyToProxy() && responseCode == HTTP_UNAUTHORIZED)
+        ) {
+            response = removeCacheEntry(chain, key, request, response);
+        }
+        return response;
+    }
+
+    private Response removeCacheEntry(Chain chain, String key, Request request, Response response) throws IOException {
+        // Remove cached authenticator and resend request
+        if (authCache.remove(key) != null) {
+            response.body().close();
+            Platform.get().log("Cached authentication expired. Sending a new request.", Platform.INFO, null);
+            // Force sending a new request without "Authorization" header
+            //interceptor at the proxy level is a Network Interceptor which does not permit to call proceed more than once.
+            if (!cacheKeyProvider.applyToProxy()) {
```

Comment:
hmm, how does proxy auth work in that case then?
Also as there is quite a bit of new code, I would appreciate if you could add a test case for the combined use of proxy and website authenticators.

Reply:
I've included 2 unit tests ('basic' auth for proxy and web site, and 'digest' auth for proxy and web site). They don't require a manual setup (i've included wiremock  which simulate proxy and web site).

These tests are coming from my own repository, which use your great library (https://github.com/clescot/kafka-connect-http).
I've also migrated from junit4 to junit 5 the current tests.
I hope you will see this initiative as positive.
best regards,
Charles.